torchtune/utils/_distributed.py

```diff
@@ -8,13 +8,21 @@
 import logging
 import os
 from itertools import chain
-from typing import Callable, Dict, Set, Tuple, Type
+from typing import Any, Callable, Dict, Set, Tuple, Type
 
 import torch
 import torch.distributed as dist
 from torch import nn
+
+try:
+    from torch.distributed._composable.fsdp import FSDPModule
+except ImportError:
+    from torch.distributed._composable.fsdp import FSDP as FSDPModule  # noqa: N811
```

Comment:
(In reference to the previous resolved conversation) To make sure I'm understanding correctly, is the try/except related to FSDP1 vs FSDP2? Is there a clear branch cut between the two versions? One other thing that could be tricky about an explicit version check is that I guess these imports will be executed anytime we import (i.e. even for recipes that have less restrictive version requirements). And I don't see a clear way to e.g. gate inside a function since the module is used as a type-hint in the function signature. So for that reason I feel like try/except might be the way to go here

Reply:
> is the try/except related to FSDP1 vs FSDP2

try/except is for FSDP2, where we renamed `FSDP` to `FSDPModule` in `torch.distributed._composable.fsdp`

Reply:
I thought we could remove this `try: ... except: ...` since we need newer PyTorch version than when we renamed it anyway.

Reply:
> I thought we could remove this try: ... except: ... since we need newer PyTorch version than when we renamed it anyway.

Oh yeah but this file may still be imported in other recipes with less strict version requirements (e.g. our full finetune distributed recipe). So in that case we could run into trouble. (Ultimately we can eventually restrict to >= 2.4 across the board, but idk that we're there yet)